python/paddle/distributed/auto_parallel/api.py

```diff
@@ -90,9 +91,324 @@ def sharding_specs(self):
+            ...         self.linear_0.weight = dist.shard_tensor(
+            ...             self.linear_0.weight,
+            ...             self._mesh,
+            ...             [Shard(1)],
+            ...             stop_gradient=False,
+            ...         )
+            ...         self.linear_1.weight = dist.shard_tensor(
+            ...             self.linear_1.weight,
+            ...             self._mesh,
+            ...             [Shard(0)],
+            ...             stop_gradient=False,
+            ...         )
+            ...     def forward(self, x):
+            ...         out = self.linear_0(x)
+            ...         out = self.relu(out)
+            ...         out = self.linear_1(out)
+            ...         return out
+
+            >>> # doctest: +REQUIRES(env:DISTRIBUTED)
+            >>> images = np.random.rand(BATCH_SIZE, IMAGE_SIZE).astype('float32')
+            >>> labels = np.random.rand(BATCH_SIZE, CLASS_NUM).astype('float32')
+            >>> dataset = RandomDataset(images, labels, BATCH_SIZE)
+            >>> loader = paddle.io.DataLoader(dataset, batch_size=BATCH_SIZE)
+
+            >>> mesh = dist.ProcessMesh([0, 1], dim_names=["x"])
+            >>> layer = DemoNet(mesh)
+            >>> opt = paddle.optimizer.SGD(
+            ...     learning_rate=0.1, parameters=layer.parameters()
+            ... )
+            >>> loss_fn = nn.MSELoss()
+
+            >>> dist_model, dist_loader = dist.to_static(
+            ...     layer, loader, loss_fn, opt
+            ... )
+
+            >>> # training
+            >>> dist_model.train()
+            >>> for batch_id, (image, label) in enumerate(dist_loader()):
+            ...     # in train mode, executing the __call__ method will
+            ...     # update the parameters of the model and return the
+            ...     # loss
+            ...     loss = dist_model(image, label)
+
+            >>> # evaluation
+            >>> dist_model.eval()
+            >>> for batch_id, (image, label) in enumerate(dist_loader()):
+            ...     # in eval mode, executing the __call__ method will
+            ...     # return the loss
+            ...     loss = dist_model(image, label)
+
+            >>> # prediction
+            >>> dist_model.predict()
+            >>> for batch_id, (image, label) in enumerate(dist_loader()):
+            ...     # in predict mode, executing the __call__ method will
+            ...     # return a dict that contains the outputs of the model,
+            ...     # where the value of "out0" is the first output.
+            ...     outs = dist_model(image)
+
+            >>> # This case need to be excuted in multi-card environment
+            >>> # export CUDA_VISIBLE_DEVICES=0,1
+            >>> # python -m paddle.distributed.launch {test_case}.py
+    """
+    dist_model = DistModel(layer, loader, loss, optimizer, strategy)
+    dist_loader = dist_model.dist_loader
+
```

Comment:
the API name  `to_static` made me very confused, especially trying very hard to understand the relation to `paddle.jit.to_static`.

from this 2 LOC implementation, I think this API basically is a creator API for `DistModel`. I'd like to suggest a more intuitive API name, e.g.: dist_model_creator.